datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -835,40 +835,38 @@ impl<'a, S: SimplifyInfo> TreeNodeRewriter for Simplifier<'a, S> {
                 left,
                 op: Or,
                 right,
-            }) if expr_contains(&left, &right, Or) => Transformed::yes(*left),
+            }) if expr_contains(&left, &right, Or)? => Transformed::yes(*left),
             // A OR (..A..) --> (..A..)
             Expr::BinaryExpr(BinaryExpr {
                 left,
                 op: Or,
                 right,
-            }) if expr_contains(&right, &left, Or) => Transformed::yes(*right),
+            }) if expr_contains(&right, &left, Or)? => Transformed::yes(*right),
             // A OR (A AND B) --> A
             Expr::BinaryExpr(BinaryExpr {
                 left,
                 op: Or,
                 right,
-            }) if is_op_with(And, &right, &left) => Transformed::yes(*left),
+            }) if is_op_with(And, &right, &left)? => Transformed::yes(*left),
             // (A AND B) OR A --> A
             Expr::BinaryExpr(BinaryExpr {
                 left,
                 op: Or,
                 right,
-            }) if is_op_with(And, &left, &right) => Transformed::yes(*right),
+            }) if is_op_with(And, &left, &right)? => Transformed::yes(*right),
             // Eliminate common factors in conjunctions e.g
             // (A AND B) OR (A AND C) -> A AND (B OR C)
             Expr::BinaryExpr(BinaryExpr {
                 left,
                 op: Or,
                 right,
-            }) if has_common_conjunction(&left, &right) => {
+            }) if has_common_conjunction(&left, &right)? => {
```

Comment:
This doesn't seem correct. It is ok to just check the presence of common non-volatile expressions in `has_common_conjunction()` to trigger simplification, but you need to deal with volatile expressions during simplification too and don't extract them as common.

E.g. can you please add a test like `column1 = 2 AND random() = 0 OR column1 = 2 AND random() = 0`?

Reply:
I checked postgresql and duckdb, they seems to merge the volatile expressions
![image](https://github.com/user-attachments/assets/30fcf9b0-726f-4f19-998a-b297a2b4ef39)
<img width="945" alt="image" src="https://github.com/user-attachments/assets/1b838b93-04e2-40ed-8bf5-4e225df0c7a1">